### PR TITLE
iTeam Support for Resolve3

### PR DESCRIPTION
**Pull request by maxtaco**
This is my picking up @modalduality's PR, but I'm not going to have enough time to get it into the code with a test. So we have to do what we can do.

- A new assertion type so that `iteam:` can make its way through the resolution code
- Testing around that
- Fixing up the resolve machinery for all of this so that we're properly reading the conflict info from the server

What remains todo:
- Formatting the display name properly for teams with conflicts
- Tests of teams and teams with conflicts.

This last part of the PR is the real doozy, since we'll have to make conflict cases work in test, and then test that this PR is going the right thing for them

### CODE REVIEW

File: go/libkb/assertion.go [hunk 1]
```diff
@@ -443,10 +471,79 @@ func (a AssertionTeamID) CheckAndNormalize(_ AssertionContext) (AssertionURL, er
...
 	var err error
 	a.name, err = keybase1.TeamNameFromString(a.Value)
+	if a.IsImplicitTeamName() {
+		return nil, fmt.Errorf("Parsed as implicit team name, wanted explicit team name.")
+	}
 	a.Value = a.name.String()
 	return a, err
 }
 
+var iTeamRxx = regexp.MustCompile(`^(?P<publicOrPrivate>(public|private))/(?P<name>[0-9a-zA-Z_,-]+)(?P<conflict> \(conflicted copy( (?P<date>\d{4}-\d{2}-\d{2}))?( #(?P<gen>\d+))?\))?$`)
```
Comment (mlsteele) [inline go/libkb/assertion.go]: Need `@` and possibly `:` for social assertions?

Comment (maxtaco) [inline go/libkb/assertion.go]: Right, and `#` for read-only lists.

File: go/libkb/resolve.go [hunk 2]
```diff
@@ -16,17 +16,18 @@ import (
...
+	uid                      keybase1.UID
+	teamID                   keybase1.TeamID
+	body                     *jsonw.Wrapper
+	err                      error
+	queriedKbUsername        string
+	queriedByUID             bool
+	resolvedKbUsername       string
+	queriedByTeamID          bool
+	resolvedTeamName         keybase1.TeamName
+	resolvedImplicitTeamName string
```
Comment (mlsteele) [inline go/libkb/resolve.go]: display name?

Comment (maxtaco) [inline go/libkb/resolve.go]: yeah, i didn't wind up populating this

**Comment by maxtaco** [comment]
We're not going to do this, we have other tickets that do more of the flow inside of the service.

**Pull request closed without merging**